### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Lessorg**

### CODE REVIEW

File: src/main/java/test/project/bookshop/dto/BookDto.java [hunk 1]
```diff
@@ -8,9 +10,14 @@
 @Setter
 public class BookDto {
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/dto/BookDto.java]: don't need to validate response

File: src/main/java/test/project/bookshop/dto/BookRequestDto.java [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Getter;
 import lombok.Setter;
 
 @Getter
 @Setter
 public class BookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/dto/BookRequestDto.java]: for String use

```suggestion
    @NotBlank
```

File: src/main/java/test/project/bookshop/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            HttpStatusCode status,
+            WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

File: src/main/java/test/project/bookshop/exception/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/exception/CustomGlobalExceptionHandler.java]: remove empty line

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**